pkg/cloud/services/network/natgateways.go

```diff
@@ -139,16 +139,30 @@ func (s *Service) deleteNatGateways() error {
 		return err
 	}
 
+	var ngIDs []*ec2.NatGateway
 	for _, sn := range s.scope.Subnets().FilterPublic() {
 		if sn.ID == "" {
 			continue
 		}
 
 		if ngID, ok := existing[sn.ID]; ok {
+			ngIDs = append(ngIDs, ngID)
+		}
+	}
+
+	c := make(chan error, len(ngIDs))
+
+	for _, ngID := range ngIDs {
+		go func(c chan error, ngID *ec2.NatGateway) {
 			err := s.deleteNatGateway(*ngID.NatGatewayId)
-			if err != nil {
-				return err
-			}
+			c <- err
+		}(c, ngID)
+	}
+
+	for i := 0; i < len(ngIDs); i++ {
+		ngwErr := <-c
+		if ngwErr != nil {
+			return err
```

Comment:
Instead of returning on the first error, how about we combine them and return all?